### PR TITLE
[5.6] Add additional collection tests

### PR DESCRIPTION
**Pull request by mnabialek**

### CODE REVIEW

File: tests/Support/SupportCollectionTest.php [hunk 1]
```diff
@@ -2519,6 +2520,35 @@ public function testGetNestedValue()
...
+        $collection->put('foo', 3);
+        $this->assertSame(['foo' => 3, 'bar' => ['nested' => 'two']], $collection->toArray());
+    }
+
+    public function testItThrowsExceptionWhenTryingToAccessNoProxyProperty()
+    {
+        $collection = new Collection();
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage('Property [foo] does not exist on this collection instance.');
+        $collection->foo;
```
Comment (Dylan-DPC) [inline tests/Support/SupportCollectionTest.php]: you are just testing if PHP works here

**Review by Dylan-DPC-zz — APPROVED** [review]

File: tests/Support/SupportCollectionTest.php (hunk 1 above)
Comment (mnabialek) [reply tests/Support/SupportCollectionTest.php]: Not really. This exception is thrown inside collection method and it should be tested whether it has been really thrown or not - now if you remove throwing exception from this method tests will pass and they shouldn't

Comment (Dylan-DPC) [reply tests/Support/SupportCollectionTest.php]: you are testing if PHP throws an exception or not when you access a non-existing property. When will this fail?

Comment (mfn) [reply tests/Support/SupportCollectionTest.php]: This isn't a PHP exceptions, it's thrown from within `\Illuminate\Support\Collection::__get`. The test is correct.

Comment (mnabialek) [reply tests/Support/SupportCollectionTest.php]: Look at implementation:

    public function __get($key)
    {
        if (! in_array($key, static::$proxies)) {
            throw new Exception("Property [{$key}] does not exist on this collection instance.");
        }

        return new HigherOrderCollectionProxy($this, $key);
    }

so I'm testing if this `if` is added and working as expected. Now there is no such test. So I'm not testing PHP, I'm testing whether this code exists and is executed as expected in this method

Comment (Dylan-DPC) [reply tests/Support/SupportCollectionTest.php]: Well yes but I still don't see the point. If this changes, other tests which use `$foo->bar` will fail as well so it is not going to be missed.

**Pull request merged**